thread/thread.cpp

```diff
@@ -648,42 +645,38 @@ namespace photon
         to->get_vcpu()->switch_count++;
     }
 
-    static void _photon_thread_die(thread* th) asm("_photon_thread_die");
-
 #if defined(__x86_64__)
 #if !defined(_WIN64)
     asm(
 DEF_ASM_FUNC(_photon_switch_context) // (void** rdi_to, void** rsi_from)
 R"(
+        rdfsbase %rax
```

Comment:
Save and restore fs register as part of context switching. It points to thread control block (TCB) and thread-local storage (TLS)